app/ami.py

```diff
@@ -134,8 +134,8 @@ def terminate_resource(self, dryrun: bool) -> bool:
                     snapshot.delete()  # delete() returns None
                 except Exception as e:
                     print(f'Failure when calling snapshot.delete(): {str(e)}')
-                    snapshots_deleted = False  # set to False and continue attempting to delete remaining Snapshots
-        return snapshots_deleted
+                    errors += f"{str(e)}\n"
+        return errors if errors.strip() else True
```

Comment:
Instead of returning true or a string, you might consider returning a list of errors. If the list is empty then the caller knows that the call must have succeeded without any errors